### PR TITLE
add route overview

### PR DESCRIPTION
**Pull request by abhishek1508**
### Description

Fixes [#885](https://github.com/mapbox/navigation-sdks/issues/885)

### Changelog

```
<changelog>Introduce route overview button.</changelog>
```

### CODE REVIEW

**Review by Guardiola31337 — COMMENTED** [review]
Testing this found some weird behaviors with the animation 👀 

overview_button

It also seems that only works the very first time i.e. panning the map and re-click does nothing.

Also found that the `OVERVIEW` text is cut in my device 👀 

overview_text_cut

**Comment by abhishek1508** [comment]
> It also seems that only works the very first time i.e. panning the map and re-click does nothing.

Doesn't happen to me cc @Guardiola31337

**Comment by abhishek1508** [comment]
> Does the overview button need to have the text? It makes sense for a muted/unmuted button that has different states, but should that also be the case for an overview that is only one action?
> 
> /cc @d-prukop

Yeah, I think it makes sense to show this optional text, because we had customers who wanted the text because they didn't know what these buttons are for. So this API in the `MapboxRouteOverviewButton` is optional. If your requirement is not to show it, don't invoke the API. You could just set the click listener on the button and change the camera angles and not invoke the API.

This API is best served when you show the button for the first time, then you can invoke the API and the user can see the text. Next time onwards don't show it.

Same applies to `Recenter Button`

cc @LukasPaczos @d-prukop

File: libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/view/MapboxRouteOverviewButton.kt [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+import com.mapbox.navigation.ui.utils.internal.extensions.shrink
+import com.mapbox.navigation.ui.utils.internal.extensions.slideWidth
+
+/**
+ * Default view to allow user to switch to route overview mode.
+ * @property textWidth Int
+ * @property binding MapboxRouteOverviewLayoutBinding
+ * @constructor
+ */
+class MapboxRouteOverviewButton @JvmOverloads constructor(
```
Comment (Guardiola31337) [inline libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/view/MapboxRouteOverviewButton.kt:23]: https://github.com/mapbox/mapbox-navigation-android/issues/4134

**Review by Guardiola31337 — APPROVED** [review]
overview_button_route_not_properly_centered

Noting that route overview is not implemented properly in the example as my current location and some part of the route is hidden by the maneuver view and the trip progress view.

**Comment by abhishek1508** [comment]
> Noting that route overview is not implemented properly in the example as my current location and some part of the route is hidden by the maneuver view and the trip progress view.

Thanks for pointing it out @Guardiola31337. This issue is more related to how the `edgeInsets` have been defined in the `MapboxNavigationActivity`. It will be fixed in a separate PR.

File: libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/view/MapboxRouteOverviewButton.kt (hunk 1 above)
Comment (abhishek1508) [reply libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/view/MapboxRouteOverviewButton.kt]: fixed

**Comment by LukasPaczos** [comment]
@abhishek1508 could you add tests for this feature?

**Pull request merged**